ftests/core/src/main/java/org/commonjava/indy/ftest/core/content/HostedContentIndexRescanTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.ftest.core.content;
+
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.client.core.module.IndyMaintenacneClientModule;
+import org.commonjava.indy.ftest.core.AbstractContentManagementTest;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.pkg.maven.model.MavenPackageTypeDescriptor;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class HostedContentIndexRescanTest
+        extends AbstractContentManagementTest
+{
+    @Test
+    public void rescanHostedAndReindexContent()
+            throws Exception
+    {
+        final String hostedName = "hosted";
+        final String path1 = "org/for/bar/1.0/foo-bar-1.0.txt";
+        final String path2 = "org/for/bar/2.0/foo-bar-2.0.txt";
+        final String path3 = "org/for/bar/3.0/foo-bar-3.0.txt";
+        final String content1 = "content1";
+        final String content2 = "content2";
+        final String content3 = "content3";
+
+        HostedRepository hosted = new HostedRepository( MavenPackageTypeDescriptor.MAVEN_PKG_KEY, hostedName );
```

Comment:
you may need to enable authoritative indexing in the Indy config files (`writeTestConfig() method override), and configure this hosted repository to use authoritative indexing. Otherwise, you'll be getting results below that are coming from the filesystem, not just the content index.